actions/build/expansion.py

```diff
@@ -47,16 +44,16 @@ def __init__(self, ai):
         return False
 
     async def handle(self, iteration):
-        if self.worker_to_first_base:
-            self.ai.actions.append(await self.send_worker_to_next_expansion())
-            self.worker_to_first_base = False
+        # if self.worker_to_first_base:
+        #     self.ai.actions.append(await self.send_worker_to_next_expansion())
+        #     self.worker_to_first_base = False
 
-        if self.expand_now:
-            await self.ai.expand_now()
-            return True
+        # if self.expand_now:
+        #     await self.ai.expand_now()
+        #     return True
 
         for expansion in self.ai.ordered_expansions:
-            if await self.ai.can_place(HATCHERY, expansion):
+            if self.ai.can_afford(HATCHERY) and await self.ai.can_place(HATCHERY, expansion):
```

Comment:
this should be checked in the `should_handle` method of the action

Reply:
ok, i didnt see it there

Reply:
https://github.com/Matuiss2/Greedy_bot/pull/87/files/99e8572c5e99563a57149b67825a90484e253a8e#diff-565218554db0ff459e1a90b6eaba600fR20